### ISSUE TITLE
Can't find application start page

### ISSUE THREAD

**Issue by jpiquot** [OrchardCMS/OrchardCore#1921]
Hi,
It seems that the last changes have broken application pages localization. My home page located in App/Areas/Framework/Views/Home/Index.cshtml is not found anymore. This was working well Friday but breaks now with the following message :

```
An unhandled exception occurred while processing the request.
InvalidOperationException: The view 'Index' was not found. The following locations were searched:
/.Modules/InspiniaTheme/Views/Framework/Home/Index.cshtml
/.Modules/InspiniaTheme/Views/Framework/Shared/Index.cshtml
/.Modules/InspiniaTheme/Views/Shared/Index.cshtml
/.Modules/Framework/Views/Home/Index.cshtml
/.Modules/Framework/Views/Shared/Index.cshtml
/Areas/Framework/Views/Home/Index.en-US.cshtml
/Areas/Framework/Views/Home/Index.en.cshtml
/Areas/Framework/Views/Home/Index.cshtml
/Areas/Framework/Views/Shared/Index.en-US.cshtml
/Areas/Framework/Views/Shared/Index.en.cshtml
/Areas/Framework/Views/Shared/Index.cshtml
/Views/Shared/Index.en-US.cshtml
/Views/Shared/Index.en.cshtml
/Views/Shared/Index.cshtml
Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable<string> originalLocations)

```
Regards,

**Comment by jtkech** [comment]
It seems that you defined a controller view at the app level, hmm, never tried to do this by using the `Areas` folder ... Now the application is implemented as a module which is always enabled so that you can define in a simpler way controller views (and razor pages) at the application level.

So, the app is now a module whose virtual subpath is `.Modules/YourAppName`. So, just try to write in the usual way your controller in your application. And then just put your view under your app in `/Views/ControllerName/Action.cshtml`.

**Comment by jpiquot** [comment]
Hi @jtkech ,
Thank you for your quick answer. I have a common home controller in the Framework module that is used by more than one application. The layout and actions of this page are common, only the static content changes from one application to the other. Until now having the controller in the Framework module and the cshtml file in the application area was a simple way to handle the first page (home page) of my business applications (not web sites). Now i have to add an unnecessary empty controller inherited from my framework and routing. It's not a lot of work but I don't like having duplicate code ;)

Thanks to the Orchard Team for all the work that has been done. Orchardcore is a great CMS, and I have learned many thing will reading the source code.

**Comment by jtkech** [comment]
Okay, i got it.

So, for this specific scenario we would just need to add this pattern through our locations expanders.

    // {2} = area = your module name
    // {1} = controller name, {0} = action name
    // {AppModuleId} = application assembly name

    /.Modules/{AppModuleId}/Areas/{2}/Views/{1}/{0}.cshtml

Then, for perf reasons we only search views defined by the app module under `Pages` and `Views`, not `App_Data` which may have many sites, under bin and obj while in dev ... So here we would just need to also search under `Areas`. Thanks for having pointed out this use case, i will think about it.

**Comment by psijkof** [comment]
@jtkech Also the Pages\Shared is not searched for partials, while in default Razor Pages app, it is. Is this an easy fix?

**Comment by jtkech** [comment]
@psijkof already fixed, i will do a PR soon.

In fact it searches in the `pageRootDirectory/Shared` and in our case we set `pageRootDirectory` to `/` to allows the razor engine to lookup for pages also under the virtual `.Modules` folder.

So, in our case it also searches under `/Shared/Foo.cshtml`, but i prefer that, for a given module (including the application), it also searches in its `/Pages/Shared` folder, not in other modules to not mix things. But i think i will allow that any module page could also find the view under the application `/Pages/Shared`.

Then if theming, i will also add the ability that a page of any module could also find a view in `TheTheme/Pages/Shared`. Hmm, then because the application is a super theme, i think it will also find it under the application `/Pages/Shared`, i will check it but it's redundant with the above.

I will do a PR soon where i will also fix `AddModularPageRoute()` and `AddModularFolderRoute()`. I broke them when implementing `@page "{Name?}"` and `[FromRoute]`, they don't work if we re-use only `@page`.

Finally, i will also think about the search in the `Areas` folder related to the original issue.
